src/scheduler/scheduler.service.ts

```diff
@@ -0,0 +1,82 @@
+import { Injectable } from '@nestjs/common';
+import { SchedulerRegistry } from '@nestjs/schedule';
+import { CronJob } from 'cron';
+import moment from 'moment';
+import { RecurringInterval, Recursion, RecursionPeriod, Schedule } from './interface/scheduler.interface';
+
+@Injectable()
+export class SchedulerService {
+  constructor(private schedulerRegistry: SchedulerRegistry) {}
+  scheduleRecurringJob(schedule: Schedule, callback: () => void) {
+    if (schedule.startDate.getTime() < Date.now()) throw new RangeError('Invalid Date: date should be after present');
+    const job = this.addRecurringJob(schedule, callback);
+    if (schedule.recursion?.period === RecursionPeriod.WEEK && schedule.recursion.repeat > 1) {
```

Comment:
note: cron doesn't have expression for every x week where x is more than 1 so I need to do it manually.